src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,189 @@
+            current = current.next;
+        }
+        return current.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Incorrect index " + index);
+        }
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        T tmp = current.value;
+        current.value = value;
+        return tmp;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Incorrect index " + index);
+        }
+        T tmp;
+        if (index == 0) {
+            tmp = head.value;
+            if (size == 1) {
+                head = null;
+                tail = null;
+            } else {
+                head = head.next;
+                head.prev = null;
+            }
+        } else if (index == size - 1) {
+            tmp = tail.value;
+            tail = tail.prev;
+            if (tail == null) {
+                head = null;
+            } else {
+                tail.next = null;
+            }
+        } else {
+            Node<T> current = head;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+            tmp = current.value;
+            current.prev.next = current.next;
+            current.next.prev = current.prev;
+        }
+        size--;
+        return tmp;
     }
 
     @Override
     public boolean remove(T object) {
+        if (head == null) {
+            return false;
+        }
+        if (head.value.equals(object)) {
+            if (size == 1) {
+                head = null;
+                tail = null;
+            } else {
+                head = head.next;
+                head.prev = null;
+            }
+            size--;
+            return true;
+        }
+        Node<T> current = head;
+        while (current != null) {
+            if ((current.value != null && current.value.equals(object))
+                    || (object == current.value)) {
+                if (current == tail) {
+                    tail = tail.prev;
+                    tail.next = null;
+                } else {
+                    current.prev.next = current.next;
+                    current.next.prev = current.prev;
+                }
+                size--;
+                return true;
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return head == null;
+    }
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
```

Comment:
Don't use getters and setters in class Node. We have access to private fields of the inner class, so using getters or setters is redundant.